sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateAsyncClientTest.java

```diff
@@ -0,0 +1,1035 @@
+import static org.mockito.Mockito.when;
+
+public class CertificateAsyncClientTest extends CertificateClientTestBase {
+    private CertificateAsyncClient certificateAsyncClient;
+
+    @Override
+    protected void beforeTest() {
+        beforeTestSetup();
+    }
+
+    private void createCertificateAsyncClient(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion, null);
+    }
+
+    private void createCertificateAsyncClient(HttpClient httpClient, CertificateServiceVersion serviceVersion,
+                                         String testTenantId) {
+        HttpPipeline httpPipeline = getHttpPipeline(httpClient, testTenantId);
+        certificateAsyncClient = spy(new CertificateClientBuilder()
+            .vaultUrl(getEndpoint())
+            .pipeline(httpPipeline)
+            .serviceVersion(serviceVersion)
+            .buildAsyncClient());
+
+        if (interceptorManager.isPlaybackMode()) {
+            when(certificateAsyncClient.getDefaultPollingInterval()).thenReturn(Duration.ofMillis(10));
+        }
+
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createCertificate(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+
+//        deleteSecretRunner((secretToDelete) -> {
+//            StepVerifier.create(secretAsyncClient.setSecret(secretToDelete))
+//                .assertNext(response -> assertSecretEquals(secretToDelete, response))
+//                .verifyComplete();
+//
+//            PollerFlux<DeletedSecret, Void> poller = secretAsyncClient.beginDeleteSecret(secretToDelete.getName());
+//            AsyncPollResponse<DeletedSecret, Void> lastResponse =
+//                poller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
+//                    .blockLast();
+//            DeletedSecret deletedSecretResponse = lastResponse.getValue();
+//
+//            assertNotNull(deletedSecretResponse.getDeletedOn());
+//            assertNotNull(deletedSecretResponse.getRecoveryId());
+//            assertNotNull(deletedSecretResponse.getScheduledPurgeDate());
+//            assertEquals(secretToDelete.getName(), deletedSecretResponse.getName());
+//        });
+
+        createCertificateRunner((certificatePolicy) -> {
+            String certName = testResourceNamer.randomName("testCert", 25);
+            PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> certPoller =
+                certificateAsyncClient.beginCreateCertificate(certName, certificatePolicy);
+
+            AsyncPollResponse<CertificateOperation, KeyVaultCertificateWithPolicy> lastResponse =
+                certPoller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
+                    .blockLast();
```

Comment:
Would something like this work?
```java
StepVerifier.create(certPoller.last().flatMap(AsyncPollResponse::getFinalResult))
    .assertNext(expected -> {
        assertNotNull(expected);
        assertEquals(certName, expected.getName());
        assertNotNull(expected.getProperties().getCreatedOn());
    }).verifyComplete();
```

Reply:
I must say though, that we would need to do this for all our other Key Vault libraries as they all do the same as Vinay did here for tests involving pollers.

Reply:
@vcolin7 yeah, the codesnippet above seems to be the async equivalent (last()) of the blockLast() call

Reply:
Since, we need to do this across KV SDKs Async Tests.
Best to do it in a follow up PR, I've filed a tracking issue. https://github.com/Azure/azure-sdk-for-java/issues/31252